src/editors/object.js

```diff
@@ -1071,7 +1071,20 @@ export class ObjectEditor extends AbstractEditor {
     if (typeof this.schema.additionalProperties === 'boolean') {
       return this.schema.additionalProperties
     }
-    return !this.jsoneditor.options.no_additional_properties
+
+    if (typeof this.schema.additionalProperties === 'object' && this.schema.additionalProperties !== null) {
```

Comment:
Could you add a comment for the new code, what&why happens here.

Reply:
I added the comments to clarify the logic of the method